browser/extensions/api/brave_shields_api.cc

```diff
@@ -221,6 +221,50 @@ BraveShieldsGetCosmeticFilteringEnabledFunction::Run() {
   return RespondNow(OneArgument(std::move(result)));
 }
 
+ExtensionFunction::ResponseAction
+BraveShieldsSetCosmeticFilteringControlTypeFunction::Run() {
+  std::unique_ptr<brave_shields::SetCosmeticFilteringControlType::Params>
+    params(
+      brave_shields::SetCosmeticFilteringControlType::Params::Create(*args_));
+  EXTENSION_FUNCTION_VALIDATE(params.get());
+
+  const GURL url(params->url);
+  // we don't allow setting defaults from the extension
```

Comment:
is there a test for this?

Reply:
I didn't specifically add one, but this uses the same logic as the other `set...ControlType` functions.

Any pointers on how I might test it? It seems like all the API calls are mocked out in the JS test suites.

Reply:
so no tests use the actual extension apis? That's a bit concerning, but I guess not something that needs to be dealt with in this PR